From dcd7a96940cd467284f81b15e4a5a4befb76099e Mon Sep 17 00:00:00 2001
Message-ID: <dcd7a96940cd467284f81b15e4a5a4befb76099e.1732384228.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1732384228.git.sharlatanus@gmail.com>
References: <cover.1732384228.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 21 Nov 2024 18:53:44 +0000
Subject: [PATCH 40/46] gnu: python-stcal: Update to 1.10.0.

* gnu/packages/astronomy.scm (python-stcal): Update to 1.10.0.
[native-inputs]: Add python-setuptools and python-wheel.

Change-Id: If0da286182d27e74af1278a7dfc1e3d54f6b8d23
---
 gnu/packages/astronomy.scm | 35 +++++++++++++++++++----------------
 1 file changed, 19 insertions(+), 16 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index a83cee194b..07ed2ea727 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -5110,26 +5110,27 @@ (define-public python-stsci-stimage
 (define-public python-stcal
   (package
     (name "python-stcal")
-    (version "1.9.0")
+    (version "1.10.0")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "stcal" version))
        (sha256
-        (base32 "1n843r19zyjm14iadfbi71ixpk0jrbhaj7h3szy1yhnhrfsrkwar"))))
+        (base32 "1h0vkc3nd77qm2ph1nihpd1n7dzr3d4rw2wga6j7siqjiwmphj3g"))))
     (build-system pyproject-build-system)
     (arguments
      (list
       #:test-flags
-      #~(list "-k" (string-append
+      #~(list "-k" (string-join
                     ;; Tests requiring network access.
-                    "not test_absolute_align"
-                    " and not test_relative_align[True]"
-                    " and not test_relative_align[False]"
-                    " and not test_get_catalog"
-                    " and not test_create_catalog"
-                    " and not test_create_catalog_graceful_failure"
-                    " and not test_parse_refcat"))
+                    (list "not test_absolute_align"
+                          "test_relative_align[True]"
+                          "test_relative_align[False]"
+                          "test_get_catalog"
+                          "test_create_catalog"
+                          "test_create_catalog_graceful_failure"
+                          "test_parse_refcat")
+                    " and not "))
       #:phases
       #~(modify-phases %standard-phases
           (add-after 'unpack 'patch-slope-fitter
@@ -5149,6 +5150,14 @@ (define-public python-stcal
             (lambda _
               ;; Cython extensions have to be built before running the tests.
               (invoke "python" "setup.py" "build_ext" "--inplace"))))))
+    (native-inputs
+     (list python-cython-3
+           python-psutil
+           python-pytest
+           python-pytest-doctestplus
+           python-setuptools
+           python-setuptools-scm
+           python-wheel))
     (propagated-inputs
      (list opencv ;Provides OpenCV-Python
            python-asdf
@@ -5160,12 +5169,6 @@ (define-public python-stcal
            python-scikit-image
            python-scipy
            python-tweakwcs))
-    (native-inputs
-     (list python-cython-3
-           python-psutil
-           python-pytest
-           python-pytest-doctestplus
-           python-setuptools-scm))
     (home-page "https://github.com/spacetelescope/stcal")
     (synopsis "STScI tools and algorithms used in calibration pipelines")
     (description "STScI tools and algorithms used in calibration pipelines.")
-- 
2.46.0

